content/en/blog/falco-0-35-0/index.md

```diff
@@ -0,0 +1,109 @@
+I think this alone was worth all the effort!  
+On behalf of the whole community, thank you for your tremendous job!  
+
+Remember, the new probe has stricter kernel releases requirements; for more info, check out our [blog post](https://falco.org/blog/falco-modern-bpf/)!
+
+## Improved Falco performance
+Thanks to the collaborative effort from Melissa Kilby, Jason Dellaluce, Andrea and me, we were able to completely revamp the way that Falco detects syscalls that needs to be captured.  
+Basically, with the new **adaptive syscalls** feature, Falco will only enable syscalls that are needed to detect the ruleset it is being ran with.  
+It will also enable a bunch of syscalls that are needed for libsinsp internal state parsers, and that's it.  
+Consequently, the `-A` flag meaning was updated. By default, ie: without the flag, heavy syscalls (like I/O ones) won't be captured, even if the ruleset ships with them, and a warning is shown to the user.  
+Using the flag will instead allow Falco to capture even heavy syscalls, with no warning.
+
+Moreover, a couple of config keys are now available to further tailor Falco adaptive syscalls feature to suit your use case.  
+A related blog post will be published in the following days, stay tuned!  
+
+Greatest thing you ask me? The huge libs refactor needed for this feature is the ground work for another highly requested feature: LSM and kprobes support!  
+
+## Falco metrics
+Thanks to another collaborative effort lead by Melissa, Falco does now expose a new, experimental `metrics` feature.  
+It introduces a redesigned stats/metrics system, emitted as monotonic counters at predefined intervals (prometheus like).  
+There are multiple options available: one can enable the output of these metrics as internal metric shapshot rule, allowing them to be notified as outputs.  
+One can also choose to output metrics to a file, that is **not** rotated by Falco.  
+Moreover, there are also options to enable CPU and memory usage metrics, internal kernel event counters and libbpf stats.  
+All in all, this is a first, great step to improve Falco resources observability!  
+
+## Falco images signing
+During this release cycle, we (Luca Guerra and me) also ported Falco release CI from CircleCI to github actions.  
+Thanks to this big effort, we are now able to sign Falco images through `cosign`. Therefore, `falco`, `falco-driver-loader`, `falco-no-driver` are now signed.  
+Moreover, Falcoctl too has seen improvements in this area, therefore its images are now signed.  
+
+## Plugins workstream
+Plugin API has seen quite a big effort, mainly from Jason, for further improvements.  
+We have now got a protocol to allow plugins to access libsinsp state; moreover, they can now declare a state table that can be accessed by other plugins.  
+Finally, they are also able to push a new async event to libsinsp main loop, to store data in state tables.  
```

Comment:
```suggestion
The Plugin API has seen quite a big effort, mainly from Jason, for further improvements.

The first big change is that the plugin framework is now totally compatible with all the events supported by the Falco libraries, including all system calls and kernel events! The plugin API now shares all the event definitions of libscap and allows plugins to both produce syscall events and extract fields from them. This feature has been on big demand since the first plugin system release ([#410](https://github.com/falcosecurity/libs/issues/410), [#992](https://github.com/falcosecurity/libs/issues/992)), and opens the door to countless new opportunities for Falco extensions.

Second, plugins now have a standard way for managing and maintaining internal state. Up until now, plugins were only able to extract fields from the information available in the payloads of each event, thus being stateless components by definition. Now, plugins have defined protocol ([#991](https://github.com/falcosecurity/libs/issues/991)) for hooking into the event stream, reconstructing an internal state, and using it for extracting fields for Falco rules! Also, plugins can inject asynchronous metadata events in open data streams to notify about state transition and make them reproduceable when replaying capture files, just like it always happened with container-related events in the Falco libraries.

Last, building on top of the previous two points, plugins are now able to communicate bidirectionally with the Falco libraries and access their internal state, both in read and write modes! For instance, this enables creating plugins that extract metadata fields from syscall event streams and that have access to all the thread information reconstructed by libsinsp, also with the opportunity of enriching it dynamically at runtime. The API surface also allows cross-plugin state access, in which a plugin is capable of interacting with the state defined by other plugins. We hope the developer community will get lots of fun out of this, the possibilities are endless.

This big feature package required altering the plugin API in a way that is **incompatible** with the previous versions (the API version major has been bumped). As such, plugins released after Falco version 0.35 will not be compatible with Falco versions <= 0.34.1, and plugins released before version 0.35 will not be compatible with Falco from version 0.35 onwards. So, the **action required** for you is to **remember to also update all your plugins to the latest versions when updating Falco to v0.35**!
```